analyzers/src/SonarAnalyzer.Common/Helpers/KnownReference.Predicates.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers;
+
+public sealed partial class KnownReference
+{
+    internal static Func<AssemblyIdentity, bool> NameIs(string name) =>
```

Comment:
Does this need to be internal? For testing purposes, add more static fields with some well-known libraries (like the xunit) that will use the feature.
For some hell-versions, take a look at `TestMethodShouldContainAssertionTest` for inspiration
```suggestion
    private static Func<AssemblyIdentity, bool> NameIs(string name) =>
```

Reply:
I'd rather like to test the single elements of the "DSL" for better test isolation. Relying on `public static KnownReference XUnit_Assert { get; }` and the like is problematic as these definitions change over time and any tests that rely on it will deteriorate. I did this to improve the situation

* Encapsulate the methods in a nested internal `Predicates` class
* Mark the class with `[EditorBrowsable(EditorBrowsableState.Never)]` to suppress IntelliSense from showing the nested class as much as possible.

Reply:
The `EditorBrowsable` is too strong solution. The nested class will do the trick.